### PR TITLE
Add interval function

### PR DESCRIPTION
**Pull request by venkateshamatam**
# Description
Add interval function

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

**Review by sunitakawane — DISMISSED** [review]

File: pkg/segment/results/mresults/metricresults.go [hunk 1]
```diff
@@ -526,3 +533,25 @@ func (r *MetricsResult) FetchPromqlMetricsForUi(mQuery *structs.MetricsQuery, pq
...
 	return httpResp, nil
 }
+
+func calculateInterval(timerangeSeconds uint32) (uint32, error) {
+	// If timerangeSeconds is greater than 10 years reject the request
+	if timerangeSeconds > 10*365*24*60*60 {
+		return 0, errors.New("timerangeSeconds is greater than 10 years")
+	}
+
+	steps := []uint32{1, 5, 10, 20, 60, 120, 300, 600, 1200, 3600, 7200, 14400}
```
Comment (nkunal) [inline pkg/segment/results/mresults/metricresults.go:543]: define this as a const at the top of the file

Comment (nkunal) [reply pkg/segment/results/mresults/metricresults.go:543]: define all the intervals that are for a timerange of 10 years

File: pkg/segment/results/mresults/metricresults.go [hunk 2]
```diff
@@ -526,3 +533,25 @@ func (r *MetricsResult) FetchPromqlMetricsForUi(mQuery *structs.MetricsQuery, pq
...
+	for _, step := range steps {
+		if timerangeSeconds/step <= 360 {
+			return step, nil
+		}
+	}
+
+	// If no suitable step is found, double the last step until a suitable one is found
+	step := steps[len(steps)-1]
+	for timerangeSeconds/step > 360 {
```
Comment (nkunal) [inline pkg/segment/results/mresults/metricresults.go:552]: once all the intervals are defined, we don't need this for loop

**Review by nkunal — COMMENTED** [review]
Add UTs for all of the expected intervals and for timerange >=10 years

File: pkg/segment/results/mresults/metricresults.go [hunk 3]
```diff
@@ -526,3 +535,18 @@ func (r *MetricsResult) FetchPromqlMetricsForUi(mQuery *structs.MetricsQuery, pq
 
 	return httpResp, nil
 }
+
+func calculateInterval(timerangeSeconds uint32) (uint32, error) {
+	// If timerangeSeconds is greater than 10 years reject the request
+	if timerangeSeconds > 10*365*24*60*60 {
```
Comment (nkunal) [inline pkg/segment/results/mresults/metricresults.go:541]: maybe define this number calc as a CONST : `TEN_YEARS_IN_SECS` and use in both places

File: pkg/segment/results/mresults/metricsresults_test.go [hunk 4]
```diff
@@ -365,3 +365,42 @@ func Test_GetResults_AggFn_QuantileFloatIndex(t *testing.T) {
...
+		216000,    // 600 * 360
+		432000,    // 1200 * 360
+		1296000,   // 3600 * 360
+		2592000,   // 7200 * 360
+		5184000,   // 14400 * 360
+		10368000,  // 28800 * 360
+		20736000,  // 57600 * 360
+		41472000,  // 115200 * 360
+		82944000,  // 230400 * 360
+		165888000, // 460800 * 360
```
Comment (nkunal) [inline pkg/segment/results/mresults/metricsresults_test.go:388]: all of the comments should say what that number represents for e.g:
```
3600,  // 1 hour
7200, // 2 hours
21600, // 6 hours
...
20736000  , // 24 days 
....
```

File: pkg/segment/results/mresults/metricsresults_test.go [hunk 5]
```diff
@@ -365,3 +365,42 @@ func Test_GetResults_AggFn_QuantileFloatIndex(t *testing.T) {
...
+		315360000, // 10years in seconds should return the max interval 921600
+		315360001, // 10 years + 1 second should return an error
+	}
+
+	for i, tr := range timerangeSeconds {
+		expectedStep := uint32(0)
```
Comment (nkunal) [inline pkg/segment/results/mresults/metricsresults_test.go:394]: better name for this var `expectedInterval`

File: pkg/segment/results/mresults/metricsresults_test.go [hunk 6]
```diff
@@ -365,3 +365,42 @@ func Test_GetResults_AggFn_QuantileFloatIndex(t *testing.T) {
...
+		if i < len(steps) {
+			expectedStep = steps[i]
+		}
+		interval, err := calculateInterval(tr)
```
Comment (nkunal) [inline pkg/segment/results/mresults/metricsresults_test.go:398]: better name for this var: `actualInterval`

**Review by nkunal — APPROVED** [review]

**Pull request merged**